File: frontEnd/src/app/core/models/update-descuento-request.ts
import { TipoCampanaDescuento } from './enums/tipo-campana-descuento.enum';
import { ProductoDescuentoRequest } from './producto-descuento-request';

export interface UpdateDescuentoRequest {
  nombre: string;
  tipo: TipoCampanaDescuento;
  fechaInicio: string;
  fechaFin: string;
  productos: ProductoDescuentoRequest[];
}

File: Backend/src/main/java/com/Trabajo_Final_Beltran/dto/request/UpdateDescuentoRequest.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.Trabajo_Final_Beltran.dto.request;

import com.Trabajo_Final_Beltran.enums.TipoCampanaDescuento;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.time.LocalDate;
import java.util.List;
import lombok.*;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder

public class UpdateDescuentoRequest {
    @NotBlank @Size(max = 100)
    private String nombre;

    @NotNull
    private TipoCampanaDescuento tipo;

    @NotNull
    private LocalDate fechaInicio;

    @NotNull
    private LocalDate fechaFin;

    @NotEmpty
    @Valid
    private List<ProductoDescuentoRequest> productos;
}